src/engine/Server.cpp

```diff
@@ -665,20 +707,155 @@ Awaitable<void> Server::sendStreamableResponse(
+  } else if (containsParam(params, "action", "sparql_json_export")) {
+    mediaType = MediaType::sparqlJson;
+  } else if (containsParam(params, "action", "turtle_export")) {
+    mediaType = MediaType::turtle;
+  } else if (containsParam(params, "action", "binary_export")) {
+    mediaType = MediaType::octetStream;
+  }
+
+  std::string_view acceptHeader = request.base()[http::field::accept];
+
+  if (!mediaType.has_value()) {
+    mediaType = ad_utility::getMediaTypeFromAcceptHeader(acceptHeader);
+  }
+
+  if (!mediaType.has_value()) {
+    throw NoSupportedMediatypeError(
+        absl::StrCat("Did not find any supported media type "
+                     "in this \'Accept:\' header field: \"",
+                     acceptHeader, "\". ",
+                     ad_utility::getErrorMessageForSupportedMediaTypes()));
+  }
+  AD_CONTRACT_CHECK(mediaType.has_value());
+  return mediaType.value();
+}
+
+// ____________________________________________________________________________
+Awaitable<void> Server::processQuery(
     const ad_utility::url_parser::ParamValueMap& params, const string& query,
     ad_utility::Timer& requestTimer,
     const ad_utility::httpUtils::HttpRequest auto& request, auto&& send,
     TimeLimit timeLimit) {
-  using namespace ad_utility::httpUtils;
+  MediaType mediaType = determineMediaType(params, request);
+  LOG(INFO) << "Requested media type of result is \""
+            << ad_utility::toString(mediaType) << "\"" << std::endl;
+
+  // TODO<c++23> use std::optional::transform
+  std::optional<uint64_t> maxSend = std::nullopt;
+  auto parameterValue =
+      ad_utility::url_parser::getParameterCheckAtMostOnce(params, "send");
+  if (parameterValue.has_value()) {
+    maxSend = std::stoul(parameterValue.value());
+  }
+  // Limit JSON requests by default
+  if (!maxSend.has_value() && (mediaType == MediaType::sparqlJson ||
+                               mediaType == MediaType::qleverJson)) {
+    maxSend = MAX_NOF_ROWS_IN_RESULT;
+  }
 
-  auto sendJson =
-      [&request, &send](
-          const json& jsonString,
-          http::status responseStatus) -> boost::asio::awaitable<void> {
-    auto response = createJsonResponse(jsonString, request, responseStatus);
-    co_return co_await send(std::move(response));
-  };
+  auto queryHub = queryHub_.lock();
+  AD_CORRECTNESS_CHECK(queryHub);
+  ad_utility::websocket::MessageSender messageSender{getQueryId(request, query),
+                                                     *queryHub};
+
+  auto [cancellationHandle, cancelTimeoutOnDestruction] =
+      setupCancellationHandle(messageSender.getQueryId(), timeLimit);
+
+  // Do the query planning. This creates a `QueryExecutionTree`, which will
+  // then be used to process the query.
+  //
+  // NOTE: This should come after determining the media type. Otherwise, it
+  // might happen that the query planner runs for a while (recall that it many
+  // do index scans) and then we get an error message afterward that a
```

Comment:
I think the `NOte:` above can be removed, as you addressed and fixed it.

Reply:
The determining of the media and query planing also were in the right order before. Do you want to remove it nonetheless?